crates/biome_js_analyze/src/lint/nursery/use_vue_multi_word_component_names.rs

```diff
@@ -0,0 +1,313 @@
+/// - Uppercase letter followed by lowercase also starts a new segment (handles cases like "UIButton")
+fn is_multi_word(name: &str) -> bool {
+    let mut segments = 0u8;
+    let mut chars = name.chars().peekable();
+
+    while let Some(ch) = chars.next() {
+        match ch {
+            '-' | '_' => {
+                // Explicit separators don't count as segments themselves
+                continue;
+            }
+            _ => {
+                // Start a new segment
+                segments += 1;
+                if segments > 1 {
+                    return true;
+                }
+
+                // Skip to the end of this segment
+                let mut prev_was_upper = ch.is_ascii_uppercase();
+                while let Some(&next_ch) = chars.peek() {
+                    match next_ch {
+                        '-' | '_' => {
+                            // End of segment due to separator
+                            break;
+                        }
+                        c if c.is_ascii_uppercase() => {
+                            if !prev_was_upper {
+                                // lowercase/digit -> uppercase: start new segment
+                                break;
+                            }
+                            // Check if this uppercase is followed by lowercase (like 'B' in "UIButton")
+                            chars.next(); // consume this uppercase char
+                            if let Some(&after_upper) = chars.peek()
+                                && after_upper.is_ascii_lowercase()
+                                && prev_was_upper
+                            {
+                                // This uppercase starts a new word (UI|Button pattern)
+                                segments += 1;
+                                if segments > 1 {
+                                    return true;
+                                }
+                                prev_was_upper = true;
+                                continue;
+                            }
+                            prev_was_upper = true;
+                        }
+                        _ => {
+                            // lowercase or digit - continue in same segment
+                            chars.next();
+                            prev_was_upper = false;
+                        }
+                    }
+                }
+            }
+        }
+        if segments > 1 {
+            return true;
+        }
+    }
+    false
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_is_multi_word() {
+        assert!(is_multi_word("MyComponent"));
+        assert!(is_multi_word("my-component"));
+        assert!(is_multi_word("myComponent"));
+        assert!(is_multi_word("MyAppRoot"));
+        assert!(is_multi_word("MYComponent"));
+        assert!(is_multi_word("foo_bar")); // underscore counts as separator
+        assert!(is_multi_word("Foo_BarBaz"));
```

Comment:
We should test words that start and finish with underscore too e.g.`_foo`, `foo_`